interface.go

```diff
@@ -93,16 +113,53 @@ func (c *Client) GetInternalSizes() (int, int) {
 
 // Create an Error artifact.
 func (c *Client) CreateError(taskID, runID, name, reason, message string) error {
-	errorReq := &tcqueue.ErrorArtifactRequest{
+	errorreq := &tcqueue.ErrorArtifactRequest{
 		Expires:     tcclient.Time(time.Now().UTC().AddDate(0, 0, 1)),
 		Message:     message,
 		Reason:      reason,
 		StorageType: "error",
 	}
 
-	logger.Printf("%s", errorReq)
+	cap, err := json.Marshal(&errorreq)
+	if err != nil {
+		return newErrorf(err, "serializing json request body for createArtifact queue call during creation of error to %s/%s/%s", taskID, runID, name)
+	}
+
+	pareq := tcqueue.PostArtifactRequest(json.RawMessage(cap))
+
+	_, err = c.queue.CreateArtifact(taskID, runID, name, &pareq)
+	if err != nil {
+		return newErrorf(err, "making createArtifact queue call during error creation of %s/%s/%s", taskID, runID, name)
+	}
+
 	return nil
+}
 
+// Create a Reference artifact.
+func (c *Client) CreateReference(taskID, runID, name, url string) error {
+	refreq := &tcqueue.RedirectArtifactRequest{
+		// What?!? Why does a 302 redirect have a content-type???
+		// Since this doesn't really make any sense, we're just going to
+		// make up one which is safe
+		ContentType: "application/octet-stream",
+		Expires:     tcclient.Time(time.Now().UTC().AddDate(0, 0, 1)),
+		StorageType: "reference",
+		URL:         url,
+	}
+
+	cap, err := json.Marshal(&refreq)
+	if err != nil {
+		return newErrorf(err, "serializing json request body for createArtifact queue call during creation of reference %s/%s/%s", taskID, runID, name)
+	}
```

Comment:
You could panic here, instead of return an error, as it would indicate a code bug,.